### PR TITLE
[@types/react-table] add cellInfo typings

### PR DESCRIPTION
**Pull request by nixxquality**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/react-tools/react-table/blob/836f2560313e6eaee3e88f3ab900b53da3b27c3c/src/index.js#L505-L520

### CODE REVIEW

**Comment by nixxquality** [comment]
Is this enough? I took it from the basic example for react-table (https://react-table.js.org/#/story/readme)
```
    accessor: 'name' // String-based value accessors!
  }, {
    Header: 'Age',
    accessor: 'age',
    Cell: props => <span className='number'>{props.value}</span> // Custom cell components!
  }, {
    id: 'friendName', // Required because our accessor is not a string
```

File: types/react-table/index.d.ts [hunk 1]
```diff
@@ -1,10 +1,6 @@
 // Type definitions for react-table 6.7
 // Project: https://github.com/react-tools/react-table
-// Definitions by: Roy Xue <https://github.com/royxue>
-//                 Pavel Sakalo <https://github.com/psakalo>
-//                 Krzysztof Porębski <https://github.com/Havret>
-//                 Grzegorz Rozdzialik <https://github.com/Gelio>
-//                 Andy S <https://github.com/andys8>
+// Definitions by: Roy Xue <https://github.com/royxue>, Pavel Sakalo <https://github.com/psakalo>, Krzysztof Porębski <https://github.com/Havret>, Andy S <https://github.com/andys8>
```
Comment (nixxquality) [inline types/react-table/index.d.ts]: Not sure why this change happened? Maybe vscode or dtslint did something automatic and I didn't notice?

**Comment by andys8** [comment]
Okay, I looked into the part of the documentation [Chapter Renderers](https://react-table.js.org/#/story/readme) and tried to confirm the changes. It's not easy.

```js
{
  // Row-level props
  row: Object, // the materialized row of data
  original: , // the original row of data
  index: '', // the index of the row in the original array
  viewIndex: '', // the index of the row relative to the current view
  level: '', // the nesting level of this row
  nestingPath: '', // the nesting path of this row
  aggregated: '', // true if this row's values were aggregated
  groupedByPivot: '', // true if this row was produced by a pivot
  subRows: '', // any sub rows defined by the `subRowKey` prop

  // Cells-level props
  isExpanded: '', // true if this row is expanded
  value: '', // the materialized value of this cell
  resized: '', // the resize information for this cell's column
  show: '', // true if the column is visible
  width: '', // the resolved width of this cell
  maxWidth: '', // the resolved maxWidth of this cell
  tdProps: '', // the resolved tdProps from `getTdProps` for this cell
  columnProps: '', // the resolved column props from 'getProps' for this cell's column
  classes: '', // the resolved array of classes for this cell
  styles: '' // the resolved styles for this cell
}
```

Could you add the documentation comments, too? Take `RowInfo` as reference. 
Also double check for differences? (`expander`, `column`)

File: types/react-table/index.d.ts [hunk 2]
```diff
@@ -13,9 +9,9 @@ export type ReactTableFunction = (value?: any) => void;
 export type AccessorFunction<D = any> = (row: D) => any;
 export type Accessor<D = any> = string | string[] | AccessorFunction<D>;
 export type Aggregator = (values: any, rows: any) => any;
-export type TableCellRenderer = ((data: any, column: any) => React.ReactNode) | React.ReactNode;
+export type TableCellRenderer = ((data: CellInfo, column: any) => React.ReactNode) | React.ReactNode;
```
Comment (andys8) [inline types/react-table/index.d.ts]: Rename `data` to `cellInfo` for consistency?
```suggestion
export type TableCellRenderer = ((cellInfo: CellInfo, column: any) => React.ReactNode) | React.ReactNode;
```

File: types/react-table/index.d.ts [hunk 3]
```diff
@@ -666,6 +662,21 @@ export interface RowInfo {
...
+    pivoted: boolean;
+    expander: boolean;
+    show: boolean;
+    width: number;
+    maxWidth: number;
+    tdProps: any;
+    columnProps: any;
+    classes: string[];
+    styles: object;
+}
```
Comment (andys8) [inline types/react-table/index.d.ts]: Add comments form documentation: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/30459#issuecomment-438218366

**Review by andys8 — CHANGES_REQUESTED** [review]

**Comment by nixxquality** [comment]
I've documented the props. As for the ones without comments:

### column
I see this one in the source code, and I use it in my own code where I use react-table so I know what it is.

### expander
This is just a shortcut for `column.expander` which is true if the column is a padding column.
```js
      // If we have SubComponent's we need to make sure we have an expander column
      if (SubComponent && !expanderColumn) {
        expanderColumn = { expander: true }
        columnsWithExpander = [expanderColumn, ...columnsWithExpander]
      }
```
https://github.com/react-tools/react-table/blob/06648b257714806888c65bc4aca8d1b295859e26/src/methods.js#L53

### pivoted
This is a shortcut for `column.pivoted` which is true when the column is pivoted.
```js
        let pivotColumnGroup = {
          Header: PivotGroupHeader,
          columns: pivotColumns.map(col => ({
            ...this.props.pivotDefaults,
            ...col,
            pivoted: true,
          })),
        }
```
https://github.com/react-tools/react-table/blob/06648b257714806888c65bc4aca8d1b295859e26/src/methods.js#L177

**Comment by andys8** [comment]
Awesome :+1:

**Review by andys8 — APPROVED** [review]

**Comment by Gelio** [comment]
Hey guys, I appreciate your work on improving the typings. One question - why was I removed from the _Authors_ list? I just found out while trying to relax the HOCs typing

**Comment by nixxquality** [comment]
Man, that's really strange. I hadn't noticed before I pushed the initial commit, but the authors list had automatically been changed (by vscode or dtslint) into a single line. I didn't notice that one of the names had totally disappeared though.
I'm so sorry!

**Comment by Gelio** [comment]
@nixxquality No worries then 😄 I just made a new PR and added myself back in, so nothing is lost 👍
